library/common-library/templates/_nodeselector.tpl

```diff
@@ -0,0 +1,10 @@
+{{- /* Defines the Pod nodeSelector */ -}}
+{{- define "common.nodeSelector" -}}
+    {{- if .Values.nodeSelector -}}
+        {{- toYaml .Values.nodeSelector -}}
+    {{- else if .Values.global -}}
+        {{- if .Values.global.nodeSelector -}}
+            {{- toYaml .Values.global.nodeSelector -}}
+        {{- end -}}
+    {{- end -}}
```

Comment:
This could be merged. Take a look at this: https://github.com/newrelic/helm-charts/pull/725/files#diff-77ece6d92f1f208ef9ef4af7ab318c4ba282722510480e737f32dd0bccb10794R24-R39

Reply:
I did not merge it to allow a re-write from the values when a custom node selector is required for a particular component, but actually both use cases might be valid.

Reply:
I thought about doing that not only in `nodeSelector` but everywhere, like in `securityContext` but I realized that it would be evil because that overrides the way Helm works.

If you use `helm upgrade something . -f values-1. yaml -f values-2. yaml -f values-3. yaml` the values are always merged.

This is sugar for customers to ease the installation but I would prefer to don't do magic under the hood. WDYT?

Reply:
Maybe @roobre who approved this has something to say...

Reply:
But if you replace a values in helm you will expect that is not merged right. I think this is a matter of precedence and is not changing the behavior of Helm. On the opposite doing merge it might be the magic in this case.

Reply:
I think Helm is a lot about merging, but mostly in layers of values and not that often whithin values themselves. Do we know if there is a community precedent where `globals` are merged with non-globals? At least I haven't seen it before and I'd rather not break users expectations on this. "A little copy is better than a little unexpected behavior".

Reply:
No, if it is a map it merge them:
```
% helm create test
Creating test
% cd test
% vim f1.yaml
% sed -e s/1/2/g f1.yaml > f2.yaml
% bat f*
───────┬──────────────
       │ File: f1.yaml
───────┼──────────────
   1   │ nodeSelector:
   2   │   test-01: is-01
   3   │
───────┴──────────────
───────┬──────────────
       │ File: f2.yaml
───────┼──────────────
   1   │ nodeSelector:
   2   │   test-02: is-02
   3   │
───────┴──────────────
```
This is how it renders with on1 value file:
```
% helm template test . -s templates/deployment.yaml -f f1.yaml | tail -5
              port: http
          resources:
            {}
      nodeSelector:
        test-01: is-01
```
This is how it renders with two:
```
% helm template test . -s templates/deployment.yaml -f f1.yaml -f f2.yaml| tail -5
          resources:
            {}
      nodeSelector:
        test-01: is-01
        test-02: is-02
```

Reply:
Answering Roberto no. I don't know anything that uses this kind of global.

That is because normally projects do not have more than 2 Helm charts but this is a really complex software to deploy. These kinds of globals are not seen frequently. At least I do not know more projects that use them. But I also down known any project that has 6 helm charts on an umbrella...

As I said, this is to ease user's configuration. I would expect helm to merge so I thought that it would be better to merge this too.

If you say that you would not merge this maybe we have to revisit `securityContext` PR. WDYT?

Reply:
oh i understand know our different point of view. I see globals values like a "parent" and not like a values at the same level.
I'm fine with both, i think both covers different use cases and if we follow one some of the use cases of the other are not going to be covered and viceversa (and probably mix them doesn't make much sense). But we need to document the behavior since it could be miss-understanding.

Reply:
My main worry would be that if we set those to be merged, it might become hard to the user to "erase" the globals and revert to the defaults. Helm does provide _some_ erasing capabilities with `null` but it's [known to have limitations](https://github.com/helm/helm/issues/9136). I'd lean towards KISS here.

Reply:
I prefer as well the "if-else" behaviour as we discussed in the first PR merely because it is easier to debug and to document. But I see the point of having the merge behaviour and guillermo said. 

So far we went for the merge (labels, security context, annotations) and the append for image pull secrets. Therefore we can follow the same approach.
 - The only thing we should make sure is that with the merge we are able to overwrite completely the global, even unsetting it